app/controllers/api/v1/admin/canonical_email_blocks_controller.rb

```diff
@@ -58,6 +58,8 @@ def set_canonical_email_blocks
   end
 
   def set_canonical_email_blocks_from_test
+    raise(Mastodon::ValidationError, "Email can't be blank") if params[:email].blank?
+
     @canonical_email_blocks = CanonicalEmailBlock.matching_email(params[:email])
```

Comment:
I think it is more idiomatic to just do `params.require(:email)`, this will raise an `ActionController::ParameterMissing` exception if `email` is blank, which is caught in `Api::BaseController` to return a proper HTTP 400 reply (though the error is not translated)